### PR TITLE
Reduce svg size & lzay load github.svg and grid.svg

### PR DESCRIPTION
**Pull request by vlin**
1. fix: remove all unnecessary React imports
    Declare `plugin:react/jsx-runtime` in `.eslintrc.js`, no longer have to import `React` to write JSX.
2. perf: reduce SVG file size during build 
    Use 'vite-plugin-svgr' to optimize all the SVG files, make them smaller. 

    Both `github.svg` and `grid.svg` are packed in the index.js 
    Before:
    distassetsyear_2012-cdd7baf2 js
    After:
    distassetsyear_2012-e71b5e80 js
    Here(https://react-svgr.com/playground/) are more info about `svgr`.
3. Split out `github.svg` and `grid.svg` from `index.js` into separate packages during the build. Only import them asynchronously when rendering the `Total` view, in order to reduce resources loaded on the initial page.
    screen

### CODE REVIEW

**Comment by yihong0618** [comment]
@frostming did something similar but seems will break some vercel preview

**Review by orionna319 — APPROVED** [review]
Thank you very much for your contribution. Nice work!

**Comment by orionna319** [comment]
Is there anything else you need my help with? I can’t see the error details during vervel deployment, It should be because I don’t have permission.

**Comment by vlin** [comment]
@yihong0618 
Would you share the Vercel deployment logs? That would help troubleshoot the cause of the failure.

**Comment by yihong0618** [comment]
> @yihong0618 Would you share the Vercel deployment logs? That would help troubleshoot the cause of the failure.

image

maybe vercel problem, will also check it

**Comment by orionna319** [comment]
[Cannot install with "frozen-lockfile" on build](https://github.com/orgs/vercel/discussions/2584)
Maybe pnpm error, can you deleting the old lockfile and running pnpm install to generate a new one.
@vlin

**Comment by vlin** [comment]
> [Cannot install with "frozen-lockfile" on build](https://github.com/orgs/vercel/discussions/2584) Maybe pnpm error, can you deleting the old lockfile(`pnpm-lock.yaml`) and running pnpm install to generate a new one. @vlin

@orionna319 
thx~

@yihong0618 
all checks have passed. let's go ahead.

File: src/pages/index.tsx [hunk 1]
```diff
@@ -132,19 +132,26 @@ const Index = () => {
...
+            return;
+          }
+          locateActivity([runId]);
+          return;
+        }
+
+        const titleEl = target.querySelector('title');
+        if (titleEl) {
+          // If the runDate exists in the <title> element, it means that a date square has been clicked.
+          const [runDate] = titleEl.innerHTML.match(/\d{4}-\d{1,2}-\d{1,2}/) || [
```
Comment (yihong0618) [inline src/pages/index.tsx:154]: nice refactor

**Review by yihong0618 — APPROVED** [review]

**Comment by yihong0618** [comment]
nice

**Pull request merged**